src/corppa/utils/generate_page_set.py

```diff
@@ -0,0 +1,127 @@
+"""
+Utility for generating a PPA page set.
+
+This method takes three inputs: (1) an input csv, (2) an output csv, and
+(3) the size of the page set.
+
+The input CSV file must have the following fields:
+    * work_id: PPA work id
+    * page_start: Starting index for page range being considered for this work
+    * page_end: Ending index for page range being considered for this work
+    * poery_pages: Comma separated list of page numbers containing poetry
+
+The pages are selected as follows:
+    * First, all pages with poetry are selected
+    * Then, all remaining pages are chosen randomly (proportionately by work)
+
+The resulting output CSV file has the following fields:
+    * work_id: PPA work id
+    * page_num: Digital page number
+"""
+
+import argparse
+import csv
+import random
+import sys
+from pathlib import Path
+
+
+def get_pages(in_csv, k):
+    """
+    Using the input CSV, generate a page set with k pages such that all pages
+    with poetry are included and the remaining pages are selected randomly from
+    the remaining pages under consideration.
+
+    Note: In some cases the generated page set may not match k.
+    """
+    # Load page data
+    page_pool = {}
+    poetry_pages = {}
+    page_counter = 0
+    with open(in_csv, newline="") as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            work_id = row["work_id"]
+            if work_id not in page_pool:
+                page_pool[work_id] = {}
+            start_idx = int(row["page_start"])
+            end_idx = int(row["page_end"]) + 1
```

Comment:
FWIW, some PPA supports non-sequential page ranges. It probably doesn't matter in this case, but would be good in future to use the same `intspan` python library for consistency and simplicity.

Reply:
I'll make a comment noting this.

Reply:
I will note that this seems like a potentially large refactor. I was going off of the poem-focused PPA test set spreadsheet, which had a single range for pages to include; but we can revisit this in the future.

Reply:
I don't think it's too big of a refactor, you just parse the range with intspan and let it generate the list of numbers in range instead of doing it based on start and end - but a comment is perfect, and we can assess whether we need it for the next round, it may still not matter (or we may be able to avoid it).

Reply:
If you are making an issue or document with notes and tag me I will add a comment there